src/System.Runtime.WindowsRuntime/tests/System/IO/CreateSafeFileHandleTests.cs

```diff
@@ -0,0 +1,249 @@
+        public void FromStorageFolder_BadSharingThrowsOutOfRange()
+        {
+            IStorageFolder folder = new StorageFolderMock();
+            Assert.Equal(
+                "share",
+                Assert.Throws<ArgumentOutOfRangeException>(() => folder.CreateSafeFileHandle("Foo", FileMode.OpenOrCreate, FileAccess.ReadWrite, (FileShare)100)).ParamName);
+        }
+
+        [Fact]
+        public void FromStorageFolder_BadOptionsThrowsOutOfRange()
+        {
+            IStorageFolder folder = new StorageFolderMock();
+            Assert.Equal(
+                "options",
+                Assert.Throws<ArgumentOutOfRangeException>(() => folder.CreateSafeFileHandle("Foo", FileMode.OpenOrCreate, FileAccess.ReadWrite, FileShare.Read, (FileOptions)100)).ParamName);
+        }
+
+        [Fact]
+        public void FromStorageFolder_InheritableThrowsNotSupported()
+        {
+            IStorageFolder folder = new StorageFolderMock();
+            Assert.Throws<NotSupportedException>(() => folder.CreateSafeFileHandle("Foo", FileMode.OpenOrCreate, FileAccess.ReadWrite, FileShare.Inheritable));
+        }
+
+        [Fact]
+        public void FromStorageFolder_EncryptedThrowsNotSupported()
+        {
+            IStorageFolder folder = new StorageFolderMock();
+            Assert.Throws<NotSupportedException>(() => folder.CreateSafeFileHandle("Foo", FileMode.OpenOrCreate, FileAccess.ReadWrite, FileShare.Read, FileOptions.Encrypted));
+        }
+
+        [Fact]
+        public void FromStorageFolder_Basic()
+        {
+            StorageFolder folder = ApplicationData.Current.TemporaryFolder;
+            string filename = "FromStorageFolder_Basic_" + Path.GetRandomFileName();
+            SafeFileHandle handle = folder.CreateSafeFileHandle(filename, FileMode.CreateNew);
+            Assert.NotNull(handle);
+            Assert.False(handle.IsInvalid);
+            handle.Close();
+            File.Delete(Path.Combine(folder.Path, filename));
+        }
+
+        [Fact]
+        public void FromStorageFolder_SurfaceIOException()
+        {
+            StorageFolder folder = ApplicationData.Current.TemporaryFolder;
+            string filename = "FromStorageFolder_SurfaceIOException_" + Path.GetRandomFileName();
+            SafeFileHandle handle = folder.CreateSafeFileHandle(filename, FileMode.CreateNew);
+            Assert.NotNull(handle);
+            Assert.False(handle.IsInvalid);
+            handle.Close();
+            Assert.Contains(
+                filename,
+                Assert.Throws<IOException>(() => folder.CreateSafeFileHandle(filename, FileMode.CreateNew)).Message);
+            File.Delete(Path.Combine(folder.Path, filename));
+        }
+
+        [Fact]
+        public void FromStorageFolder_SurfaceNotFoundException()
+        {
+            StorageFolder folder = ApplicationData.Current.TemporaryFolder;
+            string filename = "FromStorageFolder_SurfaceNotFoundException_" + Path.GetRandomFileName();
```

Comment:
Can't- using WinRT here. These are going into the app temp folder and are getting cleaned out with the app though- so doesn't matter.